### ISSUE TITLE
How do I run my application on the GPU assigned by k8s

### ISSUE THREAD

**Issue by c-android** [NVIDIA/k8s-device-plugin#336]
I have a container that runs "hashcat" to benchmark GPUs and I have a node with 8 GPUs
https://hashcat.net/hashcat/
Using resource.limits, I can deploy this container in k8s and allocate 2 GPUs for it to benchmark.

'hashcat' will still "run"  on all 8 GPUs.  The usage is limited to 2 GPUs but the hashcat output is for all 8.

Is there a way to get "hashcat" to run on the 2 GPUs that k8s has assigned to it?

**Comment by klueska** [comment]
Can you show us your full pod spec.

**Comment by c-android** [comment]
That was the right answer because as I tried to strip my spec down I realized that the SecurityContext was what was causing the problem.   

For anyone else with a similar issue setting the SecurityContext to privileged will allow the containers to see all the GPUs on a host, not just the ones that Kubernetes has allocated to it.  For 'hashcat', this meant that it thought it was benchmarking all 8 GPUs instead of the 2 in the resource requests.
